includes/admin/class-sensei-extensions.php

```diff
@@ -137,6 +138,34 @@ function( $extension ) use ( $installed_plugins ) {
 		return $extensions;
 	}
 
+	/**
+	 * Get extensions page layout.
+	 *
+	 * @since 3.11.0
+	 *
+	 * @return array
+	 */
+	public function get_layout() {
+		$transient_key    = implode( '_', [ 'sensei_extensions_layout', get_locale() ] );
+		$extension_layout = get_transient( $transient_key );
+		if ( false === $extension_layout ) {
+			$raw_layout = wp_safe_remote_get(
+				add_query_arg(
+					[ 'lang' => get_locale() ],
+					self::SENSEILMS_PRODUCTS_API_BASE_URL . '/layout'
+				)
+			);
+
+			if ( ! is_wp_error( $raw_layout ) ) {
+				$json             = json_decode( wp_remote_retrieve_body( $raw_layout ) );
+				$extension_layout = isset( $json->layout ) ? $json->layout : [];
```

Comment:
We get the array from the `layout` object to save the transient only with the array, similar to the `get_extensions` method. But when offering it through the sensei internal API, we add it back, just to follow the same approach we discussed here: 81-gh-wpcomvip/a8c-senseilms-com#discussion_r623143853 to be more flexible and extensible.